File: src/2025-02-22-00-51-50--a-deepseek-r1-test-220-10-1--2-1-huoshan/java/exercises/practice/variable-length-quantity/src/main/java/VariableLengthQuantity.java
import java.util.ArrayList;
import java.util.Collections;
import java.util.List;

class VariableLengthQuantity {

    List<String> encode(List<Long> numbers) {
        List<String> result = new ArrayList<>();
        for (Long number : numbers) {
            if (number < 0 || number > 0xFFFFFFFFL) {
                throw new IllegalArgumentException("Number out of 32-bit range");
            }
            
            List<Long> chunks = new ArrayList<>();
            long num = number;
            do {
                chunks.add(num & 0x7F);
                num >>>= 7;
            } while (num != 0);
            
            Collections.reverse(chunks);
            
            for (int i = 0; i < chunks.size(); i++) {
                long chunk = chunks.get(i);
                if (i != chunks.size() - 1) {
                    chunk |= 0x80;
                }
                result.add(String.format("%02X", chunk));
            }
        }
        return result;
    }

    List<String> decode(List<Long> bytes) {
        List<String> result = new ArrayList<>();
        long current = 0;
        int count = 0;
        
        for (Long b : bytes) {
            if (b < 0 || b > 0xFF) {
                throw new IllegalArgumentException("Invalid byte value");
            }
            
            current = (current << 7) | (b & 0x7F);
            count++;
            
            if ((b & 0x80) == 0) {
                if (count > 5 || (count == 5 && current > 0xFFFF_FFFFL)) {
                    throw new IllegalArgumentException("Overflow 32-bit unsigned");
                }
                
                result.add(String.format("%08X", current));
                current = 0;
                count = 0;
            }
        }
        
        if (count > 0) {
            throw new IllegalArgumentException("Incomplete sequence");
        }
        
        return result;
    }
}

File: src/2025-02-22-00-51-50--a-deepseek-r1-test-220-10-1--2-1-huoshan/python/exercises/practice/variable-length-quantity/variable_length_quantity.py
def encode(numbers):
    encoded = []
    for num in numbers:
        if num < 0:
            raise ValueError("Negative number not allowed")
        if num > 0xFFFFFFFF:
            raise ValueError("Number exceeds 32-bit unsigned range")
        chunks = []
        if num == 0:
            chunks.append(0)
        else:
            while num > 0:
                chunks.append(num & 0x7F)
                num >>= 7
        chunks.reverse()
        for i in range(len(chunks)):
            if i != len(chunks) - 1:
                chunks[i] |= 0x80
        encoded.extend(chunks)
    return encoded


def decode(bytes_):
    numbers = []
    current = []
    for byte in bytes_:
        current.append(byte)
        if (byte & 0x80) == 0:
            num = 0
            for b in current:
                num = (num << 7) | (b & 0x7F)
                if num > 0xFFFFFFFF:
                    raise ValueError("Number exceeds 32-bit unsigned range")
            numbers.append(num)
            current = []
    if current:
        raise ValueError("incomplete sequence")
    return numbers
